### ISSUE TITLE
Responses an associative array

### ISSUE THREAD

**Issue by tylertrotter** [johntitus/node-horseman#202]
Horseman objects have an associative array called [responses](https://gist.github.com/tylertrotter/be8da3e777c16a2b631d9de9fc94f70b#file-horseman-output-js-L30).

This tripped me up and [seems to be a bad practice](http://stackoverflow.com/questions/38215276/is-there-such-a-thing-as-an-array-object-hybrid/38216798#38216798). 

I'm wondering what the rationale was of doing it this way and if it should be changed.

Thanks for considering this.

**Comment by awlayton** [comment]
I don't believe there was a reason behind making it an array instead of an object, but that was around before I started contributing. Perhaps @johntitus could say.

It seems like it should just be initialized as an object [here](https://github.com/johntitus/node-horseman/blob/master/lib/index.js#L143) instead of as an array.

**Comment by johntitus** [comment]
Huh, weird. Never noticed that before. Just a goof on my part I guess. It would seem to make more sense as object, unless, for some reason, the same object could be requested multiple times on the same page with differing response codes. I can't think of a good reason WHY that would happen, but I guess it's possible.
